cmd/subctl/deploybroker.go

```diff
@@ -43,20 +43,20 @@ var deployBroker = &cobra.Command{
 	Use:   "deploy-broker",
 	Short: "Deploys the broker",
 	Run: func(cmd *cobra.Command, args []string) {
+		status := cli.NewReporter()
+
 		config, err := restConfigProducer.ForCluster()
-		exit.OnError("Error creating REST config", err)
+		exit.OnError(status.Error(err, "Error creating REST config"))
 
 		clientProducer, err := client.NewProducerFromRestConfig(config)
-		exit.OnError("Error creating client producer", err)
+		exit.OnError(status.Error(err, "Error creating client producer"))
 
-		status := cli.NewReporter()
 		err = deploy.Broker(&deployflags, clientProducer, status)
-		if err == nil {
-			err = broker.WriteInfoToFile(config, deployflags.BrokerNamespace, ipsecSubmFile,
-				stringset.New(deployflags.BrokerSpec.Components...), deployflags.BrokerSpec.DefaultCustomDomains, status)
-		}
+		exit.OnError(err)
```

Comment:
exit.OnError(status.Error(.....))?

Reply:
`deploy.Broker` has already logged the error via the `status`. Here we just want to exit with a non-zero code if `err` is  non-nil.

Reply:
On second thought, `status` should end in the `pkg` itself and not propagate to `cmd/subctl`. This way any user using `broker` pkg will have a complete status implementation.